pkg/nsx/services/subnetport/builder.go

```diff
@@ -50,6 +72,25 @@ func (service *SubnetPortService) buildSubnetPort(obj interface{}, nsxSubnet *mo
 					MacAddress: &o.Status.NetworkInterfaceConfig.MACAddress,
 				},
 			}
+			// Chck if the specified MAC is in NSX MAC Pool to determine the allocateAddresses
+			inNSXMacPool, err := service.isInNSXMacPool(o.Status.NetworkInterfaceConfig.MACAddress)
+			if err != nil {
+				return nil, fmt.Errorf("failed to check NSX MAC Pool: %w", err)
+			}
+			isIPPool = !inNSXMacPool
+		} else if len(o.Spec.AddressBindings) > 0 {
+			addressBindings = []model.PortAddressBindingEntry{
+				{
+					IpAddress:  &o.Spec.AddressBindings[0].IPAddress,
+					MacAddress: &o.Spec.AddressBindings[0].MACAddress,
```

Comment:
we also need consider if macAddress is empty.

Reply:
Thanks, only set MacAddress when Spec.AddressBindings[0].MACAddress is not empty.